# Various examples using the NNDC query tools

First, import necessary tools.

In [1]:
import numpy as np
import pandas as pd
from becquerel.tools import nndc

## Example 1: Cesium-137

### 1a: Look up the half-life of Cesium-137
Fetch the "wallet card" data for Cs-137. The wallet card includes the Z, A, half-life, nuclear spin, etc.

In [2]:
card = nndc.fetch_wallet_card(nuc='Cs-137')
display(card)
print('')
t12 = max(card['T1/2 (s)'])
print('{} s'.format(t12))
print('{} y'.format(t12 / 3600 / 24 / 365.25))

Z Element    A m  M   N   JPi  Energy Level (MeV) Decay Mode  \
0  55      Cs  137    0  82  7/2+                 0.0         B-   

   Branching (%) Abundance (%)  Mass Excess (MeV)     T1/2 (s)    T1/2 (txt)  
0          100.0          None  -86.5456+/-0.0004  949300000.0  30.08 Y 0.09


949300000.0 s
30.08150176185768 y


### 1b: Find the branching ratio of the 662 keV gamma-ray line of Cs-137
Fetch the "decay radiation" data for the line at 662 keV. For gamma rays, the ```type``` keyword should be set to ```"Gamma"```.

In [3]:
rad = nndc.fetch_decay_radiation(nuc='Cs-137', type='Gamma',
                                 i_range=(1, None), e_range=(661, 663))
display(rad)
print('')
print('{:.2f} keV line intensity: {}%'.format(
    rad['Radiation Energy (keV)'][0],
    rad['Radiation Intensity (%)'][0]))

Z Element    A m  M   N   JPi  Energy Level (MeV) Decay Mode Radiation  \
0  55      CS  137    0  82  7/2+                 0.0         B-         G   

  Radiation subtype Radiation Energy (keV) Radiation Intensity (%) Daughter  \
0                        661.6570+/-0.0030            85.10+/-0.20    137BA   

  Dose (MeV / Bq / s) Endpoint Energy (keV)     T1/2 (s) T1/2 (txt)  
0     0.5631+/-0.0013                  None  949300000.0  30.08 Y 9


661.66+/-0.00 keV line intensity: 85.10+/-0.20%


## Example 2: Barium-133

### 2a: Fetch isotope data for Ba-133

In [4]:
card = nndc.fetch_wallet_card(nuc='Ba-133')
display(card)

Z Element    A  m  M   N    JPi  Energy Level (MeV) Decay Mode  \
0  56      Ba  133     0  77   1/2+              0.0000         EC   
1  56      Ba  133  m  1  77  11/2-              0.2883         IT   
2  56      Ba  133  m  1  77  11/2-              0.2883         EC   

   Branching (%) Abundance (%)  Mass Excess (MeV)     T1/2 (s)      T1/2 (txt)  
0         100.00          None  -87.5536+/-0.0010  333000000.0  10.551 Y 0.011  
1          99.99          None  -87.2653+/-0.0010     140148.0     38.93 H 0.1  
2           0.01          None  -87.2653+/-0.0010     140148.0     38.93 H 0.1

The basic query includes the 288 MeV excited state with a half-life of 39 hours. By using the ```elevel_range``` keyword we can query only the ground state:

In [5]:
card = nndc.fetch_wallet_card(nuc='Ba-133', elevel_range=(0, 0))
display(card)

Z Element    A m  M   N   JPi  Energy Level (MeV) Decay Mode  \
0  56      Ba  133    0  77  1/2+                 0.0         EC   

   Branching (%) Abundance (%)  Mass Excess (MeV)     T1/2 (s)      T1/2 (txt)  
0          100.0          None  -87.5536+/-0.0010  333000000.0  10.551 Y 0.011

### 2b: Find the main gamma-ray lines of Ba-133
Next, find all gamma-rays for decays from the ground state of Ba-133 that are &ge;5% intensity. Use the ```i_range``` keyword to select a range of intensities.

In [6]:
rad = nndc.fetch_decay_radiation(nuc='Ba-133', type='Gamma', i_range=(5, None))
cols = ['Z', 'Element', 'A', 'Decay Mode', 'Radiation', 'Radiation Energy (keV)',
        'Radiation Intensity (%)', 'Energy Level (MeV)']
display(rad[cols])

Z Element    A Decay Mode Radiation Radiation Energy (keV)  \
0   56      BA  133         IT         G                   4.47   
1   56      BA  133         IT         G                 31.817   
2   56      BA  133         IT         G                 32.194   
3   56      BA  133         IT         G                 36.378   
4   56      BA  133         IT         G        275.925+/-0.007   
5   56      BA  133         EC         G                   4.29   
6   56      BA  133         EC         G                 30.625   
7   56      BA  133         EC         G                 30.973   
8   56      BA  133         EC         G                  34.92   
9   56      BA  133         EC         G                 34.987   
10  56      BA  133         EC         G       80.9979+/-0.0011   
11  56      BA  133         EC         G      276.3989+/-0.0012   
12  56      BA  133         EC         G      302.8508+/-0.0005   
13  56      BA  133         EC         G      356.0129+/-0.0007   
14  56      BA  133         EC         G      383.8485+/-0.0012   

   Radiation Intensity (%)   Energy Level (MeV)  
0               16.4+/-0.8  0.288252+/-0.000009  
1               15.1+/-0.4  0.288252+/-0.000009  
2               27.6+/-0.6  0.288252+/-0.000009  
3              5.10+/-0.11  0.288252+/-0.000009  
4                    17.69  0.288252+/-0.000009  
5               15.7+/-0.8                    0  
6               33.9+/-1.0                    0  
7               62.2+/-1.8                    0  
8              5.88+/-0.17                    0  
9             11.40+/-0.30                    0  
10            32.90+/-0.30                    0  
11             7.16+/-0.05                    0  
12            18.34+/-0.13                    0  
13                   62.05                    0  
14             8.94+/-0.06                    0

Once again, this query includes the short-lived excited state. To select only the ground state, use the `elevel_range` kwarg:

In [7]:
rad = nndc.fetch_decay_radiation(nuc='Ba-133', type='Gamma', i_range=(5, None), elevel_range=(0, 0))
display(rad[cols])

Z Element    A Decay Mode Radiation Radiation Energy (keV)  \
5   56      BA  133         EC         G                   4.29   
6   56      BA  133         EC         G                 30.625   
7   56      BA  133         EC         G                 30.973   
8   56      BA  133         EC         G                  34.92   
9   56      BA  133         EC         G                 34.987   
10  56      BA  133         EC         G       80.9979+/-0.0011   
11  56      BA  133         EC         G      276.3989+/-0.0012   
12  56      BA  133         EC         G      302.8508+/-0.0005   
13  56      BA  133         EC         G      356.0129+/-0.0007   
14  56      BA  133         EC         G      383.8485+/-0.0012   

   Radiation Intensity (%) Energy Level (MeV)  
5               15.7+/-0.8                  0  
6               33.9+/-1.0                  0  
7               62.2+/-1.8                  0  
8              5.88+/-0.17                  0  
9             11.40+/-0.30                  0  
10            32.90+/-0.30                  0  
11             7.16+/-0.05                  0  
12            18.34+/-0.13                  0  
13                   62.05                  0  
14             8.94+/-0.06                  0

## Example 3: Hf-178 (multiple excited states)

Look up the half-lives for the ground state and two excited states of Hf-178:

In [8]:
card = nndc.fetch_wallet_card(nuc='Hf-178')
display(card)

Z Element    A   m  M    N  JPi  Energy Level (MeV) Decay Mode  \
0  72      Hf  178      0  106   0+              0.0000              
1  72      Hf  178  m1  1  106   8-              1.1474         IT   
2  72      Hf  178  m2  2  106  16+              2.4461         IT   

   Branching (%) Abundance (%)  Mass Excess (MeV)      T1/2 (s) T1/2 (txt)  
0            NaN  27.28+/-0.07  -52.4352+/-0.0014           inf     STABLE  
1          100.0          None  -51.2878+/-0.0014  4.000000e+00  4.0 S 0.2  
2          100.0          None  -49.9891+/-0.0014  9.783000e+08     31 Y 1

## Example 4: Uranium-235

### 4a: Find the different types of decays of U-235
This information is in the wallet card:

In [9]:
card = nndc.fetch_wallet_card(nuc='U-235', elevel_range=(0, 0))
display(set(card['Decay Mode']))

{'20Ne', '25Ne', '28Mg', 'A', 'SF'}

### 4b: Find the main alpha lines for U-235

In [10]:
rad = nndc.fetch_decay_radiation(nuc='U-235', type='Alpha', i_range=(10, None))
display(rad[cols])

Z Element    A Decay Mode Radiation Radiation Energy (keV)  \
0  92       U  235          A         A           4364.3+/-0.4   
1  92       U  235          A         A           4395.4+/-0.4   

  Radiation Intensity (%)  Energy Level (MeV)  
0            18.92+/-0.14                 0.0  
1            57.73+/-0.24                 0.0

### 4c: Find the main gamma lines for U-235

In [11]:
rad = nndc.fetch_decay_radiation(nuc='U-235', type='Gamma', i_range=(10, None))
display(rad[cols])

Z Element    A Decay Mode Radiation Radiation Energy (keV)  \
0  92       U  235          A         G                     13   
1  92       U  235          A         G           19.55+/-0.05   
2  92       U  235          A         G        143.760+/-0.020   
3  92       U  235          A         G        185.715+/-0.005   

  Radiation Intensity (%)  Energy Level (MeV)  
0                  27+/-4                 0.0  
1                      63                 0.0  
2            10.96+/-0.14                 0.0  
3              57.0+/-0.6                 0.0

## Example 5: Pa-234m

### Look up the 1001 keV gamma-ray line from the decay of Pa-234m
Requires searching for decays from an excited state of a nucleus. Use the `elevel_range` keyword argument to select parent energy levels greater than 0.

In [12]:
rad = nndc.fetch_decay_radiation(nuc='Pa-234', type='Gamma', e_range=(999, 1002), elevel_range=(0.001, None))
display(rad[cols])

Z Element    A Decay Mode Radiation Radiation Energy (keV)  \
0  91      PA  234         B-         G         1001.03+/-0.10   

  Radiation Intensity (%)  Energy Level (MeV)  
0           0.842+/-0.008             0.07392

## Example 6: Find long-lived isotopes with gamma-ray lines at 1461 keV
Are there any others besides K-40? Use the ```t_range``` keyword to look for any other isotopes with half-lives greater than 1E9 seconds.

In [13]:
rad = nndc.fetch_decay_radiation(t_range=[1e9, None], type='Gamma', e_range=[1460, 1462])
display(rad[cols])

Z Element   A Decay Mode Radiation Radiation Energy (keV)  \
0  19       K  40         EC         G       1460.820+/-0.005   

  Radiation Intensity (%)  Energy Level (MeV)  
0            10.66+/-0.17                 0.0

## Example 7: Find isotopes with lines at 661.7 keV for 120 &le; A &le; 160

This is similar to the last example, but this one shows how to use the ```a_range``` keyword.

In [14]:
rad = nndc.fetch_decay_radiation(
    a_range=[120, 160],
    e_range=[661.2, 662.2],
    i_range=[0.1, None],
    elevel_range=[0, 0], # ground states only
)
display(rad[cols])

Z Element    A Decay Mode Radiation Radiation Energy (keV)  \
0   53       I  120         EC         G          662.10+/-0.30   
1   54      XE  121         EC        BP                662+/-5   
2   55      CS  137         B-         G      661.6570+/-0.0030   
3   55      CS  143         B-         G          662.10+/-0.10   
5   57      LA  144         B-        BM               662+/-25   
6   61      PM  139         EC        BP               662+/-14   
7   61      PM  139         EC         G            661.3+/-0.4   
8   61      PM  152         B-         G            661.7+/-0.4   
9   61      PM  154         B-         G          661.70+/-0.30   
10  67      HO  157         EC         G          661.90+/-0.10   

   Radiation Intensity (%) Energy Level (MeV)  
0              1.05+/-0.04                  0  
1              0.48+/-0.08                  0  
2             85.10+/-0.20                  0  
3                3.0+/-0.5                  0  
5              0.63+/-0.13                  0  
6            0.110+/-0.030                  0  
7              0.41+/-0.08                  0  
8            0.240+/-0.030                  0  
9                     0.14                  0  
10             0.32+/-0.05                  0

## Example 8: Find long-lived isotopes with lines at 186 keV
Famously, Ra-226 and U-235 have a similar gamma-ray line here.

In [15]:
rad = nndc.fetch_decay_radiation(
    e_range=[185, 187],
    i_range=[1, None],
    t_range=[1e10, None],
    elevel_range=[0, 0], # ground states only
)
display(rad[cols])

Z Element    A Decay Mode Radiation Radiation Energy (keV)  \
0  88      RA  226          A         G        186.211+/-0.013   
1  92       U  235          A         G        185.715+/-0.005   

  Radiation Intensity (%)  Energy Level (MeV)  
0             3.64+/-0.04                 0.0  
1              57.0+/-0.6                 0.0

## Example 9: Find even- and odd-A isotopes of Hydrogen

You can use the ```a_even``` and ```a_odd``` keywords to select only even or odd ```a``` values. There are equivalents for ```z``` and ```n``` as well.

In [16]:
evens = nndc.fetch_wallet_card(z=1, a_even=True, elevel_range=(0, 0))
display(evens[['Z', 'Element', 'A', 'Abundance (%)', 'T1/2 (s)']])

Z Element  A  Abundance (%)      T1/2 (s)
0  1       H  2  0.011+/-0.007           inf
1  1       H  4           None           NaN
2  1       H  6           None  2.848000e-22

In [17]:
odds = nndc.fetch_wallet_card(z=1, a_odd=True, elevel_range=(0, 0))
display(odds[['Z', 'Element', 'A', 'Abundance (%)', 'T1/2 (s)']])

Z Element  A   Abundance (%)      T1/2 (s)
0  1       H  1  99.989+/-0.007           inf
1  1       H  3            None  3.888000e+08
2  1       H  5            None  7.995000e-23

## Example 10: Uranium-238

### 10a: Find all significant U-238 series gamma-ray lines

First we will define a function:

In [18]:
def series_radiation(isotopes, rtype='Gamma', emin=30., imin=0.7, branchings=None):
    """Return a DataFrame of all the gammas from the decay series."""
    df_series = pd.DataFrame()
    if branchings is None:
        branchings = np.ones_like(isotopes)
    assert len(isotopes) == len(branchings)
    for isotope, branching in zip(isotopes, branchings):
        print(isotope, branching)
        isotope_id = isotope
        if isotope.endswith('m'):
            isotope_id = isotope[:-1]
        df = nndc.fetch_decay_radiation(nuc=isotope_id, type=rtype, e_range=(emin, 1e4), i_range=(imin, None))
        if isotope.endswith('m'):
            df = df.loc[df['Energy Level (MeV)'] > 0.]
        else:
            df = df.loc[df['Energy Level (MeV)'] == 0.]
        df['Radiation Intensity (%)'] *= float(branching)
        if len(df) == 0:
            continue
        df_series = df_series.append(df)
    return df_series

Now let's run this function on the U-238 series:

In [19]:
series = ['U-238', 'Th-234', 'Pa-234m', 'U-234',
          'Th-230', 'Ra-226', 'Rn-222', 'Po-218',
          'Pb-214', 'Bi-214', 'Po-214', 'Pb-210',
          'Bi-210', 'Po-210']
gammas = series_radiation(series)
gammas.sort_values(by='Radiation Energy (keV)', inplace=True)
fields = ['Z', 'Element', 'A', 'Decay Mode',
          'Radiation', 'Radiation subtype',
          'Radiation Energy (keV)', 'Radiation Intensity (%)',
          'Daughter']
display(gammas[fields])

U-238 1
Th-234 1
Pa-234m 1
U-234 1
Th-230 1
Ra-226 1
Rn-222 1
Po-218 1
Pb-214 1
Bi-214 1
Po-214 1
Pb-210 1
Bi-210 1
Po-210 1


Z Element    A Decay Mode Radiation Radiation subtype  \
0   82      PB  210         B-         G                     
0   82      PB  214         B-         G                     
0   90      TH  234         B-         G                     
1   82      PB  214         B-         G            XR ka2   
2   82      PB  214         B-         G            XR ka1   
0   83      BI  214         B-         G            XR ka1   
3   82      PB  214         B-         G            XR kb3   
4   82      PB  214         B-         G            XR kb1   
5   82      PB  214         B-         G            XR kb2   
1   90      TH  234         B-         G                     
2   90      TH  234         B-         G                     
0   88      RA  226          A         G                     
6   82      PB  214         B-         G                     
7   82      PB  214         B-         G                     
8   82      PB  214         B-         G                     
1   83      BI  214         B-         G                     
2   83      BI  214         B-         G                     
3   83      BI  214         B-         G                     
9   82      PB  214         B-         G                     
4   83      BI  214         B-         G                     
5   83      BI  214         B-         G                     
0   91      PA  234         B-         G                     
6   83      BI  214         B-         G                     
7   83      BI  214         B-         G                     
8   83      BI  214         B-         G                     
9   83      BI  214         B-         G                     
10  83      BI  214         B-         G                     
11  83      BI  214         B-         G                     
12  83      BI  214         B-         G                     
13  83      BI  214         B-         G                     
14  83      BI  214         B-         G                     
15  83      BI  214         B-         G                     
16  83      BI  214         B-         G                     
17  83      BI  214         B-         G                     
18  83      BI  214         B-         G                     
19  83      BI  214         B-         G                     
20  83      BI  214         B-         G                     
21  83      BI  214         B-         G                     
22  83      BI  214         B-         G                     

   Radiation Energy (keV) Radiation Intensity (%) Daughter  
0        46.5390+/-0.0010             4.25+/-0.04    210BI  
0        53.2284+/-0.0018           1.075+/-0.007    214BI  
0          63.290+/-0.020               3.7+/-0.4    234PA  
1                  74.815             5.80+/-0.30    214BI  
2                  77.107               9.7+/-0.5    214BI  
0                   79.29           0.907+/-0.023    214PO  
3                   86.83             1.17+/-0.06    214BI  
4                  87.349             2.24+/-0.11    214BI  
5                  89.784             0.82+/-0.04    214BI  
1          92.380+/-0.010             2.13+/-0.20    234PA  
2          92.800+/-0.020             2.10+/-0.20    234PA  
0         186.211+/-0.013             3.64+/-0.04    222RN  
6       241.9950+/-0.0023           7.251+/-0.016    214BI  
7       295.2228+/-0.0018            18.42+/-0.04    214BI  
8       351.9321+/-0.0018            35.60+/-0.07    214BI  
1         609.320+/-0.005            45.49+/-0.16    214PO  
2         665.447+/-0.009           1.531+/-0.006    214PO  
3         768.360+/-0.005           4.894+/-0.011    214PO  
9           785.96+/-0.08           1.060+/-0.030    214BI  
4         806.180+/-0.009           1.264+/-0.005    214PO  
5         934.056+/-0.006           3.107+/-0.010    214PO  
0          1001.03+/-0.10           0.842+/-0.008     234U  
6        1120.294+/-0.006          14.920+/-0.030    214PO  
7        1155.210+/-0.008           1.633+/-0.006    214PO  

### 10b: Find all significant U-238 alpha lines

In [20]:
alphas = series_radiation(series, rtype='Alpha')
alphas.sort_values(by='Radiation Energy (keV)', inplace=True)
display(alphas[fields])

U-238 1
Th-234 1
Pa-234m 1
U-234 1
Th-230 1
Ra-226 1
Rn-222 1
Po-218 1
Pb-214 1
Bi-214 1
Po-214 1
Pb-210 1
Bi-210 1
Po-210 1


Z Element    A Decay Mode Radiation Radiation subtype  \
0  92       U  238          A         A                     
1  92       U  238          A         A                     
0  88      RA  226          A         A                     
0  90      TH  230          A         A                     
1  90      TH  230          A         A                     
0  92       U  234          A         A                     
1  92       U  234          A         A                     
1  88      RA  226          A         A                     
0  84      PO  210          A         A                     
0  86      RN  222          A         A                     
0  84      PO  218          A         A                     
0  84      PO  214          A         A                     

  Radiation Energy (keV) Radiation Intensity (%) Daughter  
0               4151+/-5              21.0+/-3.0    234TH  
1           4198.0+/-3.0              79.0+/-3.0    234TH  
0           4601.0+/-1.0           6.160+/-0.030    222RN  
0           4620.5+/-1.5            23.40+/-0.10    226RA  
1           4687.0+/-1.5            76.30+/-0.30    226RA  
0           4722.4+/-1.4            28.42+/-0.09    230TH  
1           4774.6+/-1.4            71.38+/-0.16    230TH  
1         4784.34+/-0.25            93.84+/-0.11    222RN  
0         5304.33+/-0.07                     100    206PB  
0         5489.48+/-0.30          99.920+/-0.010    218PO  
0         6002.35+/-0.09        99.9789+/-0.0023    214PB  
0         7686.82+/-0.07        99.9895+/-0.0006    210PB

### 10c: Find all significant Th-232 series gamma-ray lines

Unlike the U-238 series, we need to provide branching ratios to account for the decay of Bi-212 into either Po-212 or Tl-208.

In [21]:
series = ['Th-232', 'Ra-228', 'Ac-228', 'Th-228',
          'Ra-224', 'Rn-220', 'Po-216', 'Pb-212',
          'Bi-212', 'Po-212', 'Tl-208']
branchings = [1, 1, 1, 1,
              1, 1, 1, 1,
              1, 0.6406, 0.3594]
gammas = series_radiation(series, branchings=branchings)
gammas.sort_values(by='Radiation Energy (keV)', inplace=True)
fields = ['Z', 'Element', 'A', 'Decay Mode',
          'Radiation', 'Radiation subtype',
          'Radiation Energy (keV)', 'Radiation Intensity (%)',
          'Daughter']
display(gammas[fields])

Th-232 1
Ra-228 1
Ac-228 1
Th-228 1
Ra-224 1
Rn-220 1
Po-216 1
Pb-212 1
Bi-212 1
Po-212 0.6406
Tl-208 0.3594


Z Element    A Decay Mode Radiation Radiation subtype  \
0   83      BI  212          A         G                     
0   81      TL  208         B-         G            XR ka2   
0   82      PB  212         B-         G            XR ka2   
1   81      TL  208         B-         G            XR ka1   
1   82      PB  212         B-         G            XR ka1   
0   90      TH  228          A         G                     
2   81      TL  208         B-         G            XR kb1   
2   82      PB  212         B-         G            XR kb3   
3   82      PB  212         B-         G            XR kb1   
4   82      PB  212         B-         G            XR kb2   
0   89      AC  228         B-         G            XR ka2   
1   89      AC  228         B-         G            XR ka1   
2   89      AC  228         B-         G                     
3   89      AC  228         B-         G            XR kb1   
4   89      AC  228         B-         G                     
5   89      AC  228         B-         G                     
6   89      AC  228         B-         G                     
7   89      AC  228         B-         G                     
5   82      PB  212         B-         G                     
0   88      RA  224          A         G                     
3   81      TL  208         B-         G                     
8   89      AC  228         B-         G                     
4   81      TL  208         B-         G                     
6   82      PB  212         B-         G                     
9   89      AC  228         B-         G                     
10  89      AC  228         B-         G                     
11  89      AC  228         B-         G                     
12  89      AC  228         B-         G                     
5   81      TL  208         B-         G                     
13  89      AC  228         B-         G                     
6   81      TL  208         B-         G                     
14  89      AC  228         B-         G                     
1   83      BI  212         B-         G                     
15  89      AC  228         B-         G                     
7   81      TL  208         B-         G                     
16  89      AC  228         B-         G                     
2   83      BI  212         B-         G                     
17  89      AC  228         B-         G                     
18  89      AC  228         B-         G                     
19  89      AC  228         B-         G                     
8   81      TL  208         B-         G                     
20  89      AC  228         B-         G                     
21  89      AC  228         B-         G                     
22  89      AC  228         B-         G                     
23  89      AC  228         B-         G                     
24  89      AC  228         B-         G                     
25  89      AC  228         B-         G                     
26  89      AC  228         B-         G                     
3   83      BI  212         B-         G                     
27  89      AC  228         B-         G                     
9   81      TL  208         B-         G                     

   Radiation Energy (keV) Radiation Intensity (%) Daughter  
0          39.857+/-0.004             1.06+/-0.09    208TL  
0                  72.805           0.722+/-0.022    208PB  
0                  74.815            10.28+/-0.25    212BI  
1                  74.969           1.204+/-0.032    208PB  
1                  77.107              17.1+/-0.4    212BI  
0        84.3730+/-0.0030             1.19+/-0.04    224RA  
2                  84.938           0.279+/-0.008    208PB  
2                   86.83             2.07+/-0.05    212BI  
3                  87.349             3.97+/-0.09    212BI  
4                  89.784           1.460+/-0.030    212BI  
0                  89.957             1.90+/-0.30    228TH  
1                   93.35               3.1+/-0.

### 10d: Find all significant Th-232 alpha lines

In [22]:
alphas = series_radiation(series, branchings=branchings, rtype='Alpha')
alphas.sort_values(by='Radiation Energy (keV)', inplace=True)
display(alphas[fields])

Th-232 1
Ra-228 1
Ac-228 1
Th-228 1
Ra-224 1
Rn-220 1
Po-216 1
Pb-212 1
Bi-212 1
Po-212 0.6406
Tl-208 0.3594


Z Element    A Decay Mode Radiation Radiation subtype  \
0  90      TH  232          A         A                     
1  90      TH  232          A         A                     
0  90      TH  228          A         A                     
1  90      TH  228          A         A                     
0  88      RA  224          A         A                     
1  88      RA  224          A         A                     
3  83      BI  212          A         A                     
4  83      BI  212          A         A                     
0  86      RN  220          A         A                     
0  84      PO  216          A         A                     
0  84      PO  212          A         A                     

  Radiation Energy (keV) Radiation Intensity (%) Daughter  
0           3947.2+/-2.0              21.7+/-1.3    228RA  
1           4012.3+/-1.4              78.2+/-1.3    228RA  
0         5340.36+/-0.15              26.0+/-1.0    224RA  
1         5423.15+/-0.22              73.4+/-0.5    224RA  
0           5448.6+/-1.2             5.06+/-0.05    220RN  
1         5685.37+/-0.15            94.92+/-0.05    220RN  
3       6050.780+/-0.030            25.13+/-0.07    208TL  
4       6089.880+/-0.030             9.75+/-0.05    208TL  
0         6288.08+/-0.10          99.886+/-0.017    216PO  
0           6778.3+/-0.5      99.99810+/-0.00030    212PB  
0         8784.86+/-0.12                   64.06    208PB